src/WpfMath/UserDefinedColorParser.cs

```diff
@@ -0,0 +1,264 @@
+                        resultColor.R = channelbytes[0];
+                        resultColor.G = channelbytes[1];
+                        resultColor.B = channelbytes[2];
+                        break;
+                    }
+                case ColorStringTypes.Hex_longString:
+                    {
+                        byte[] channelbytes = ByteHexStringValues(input.Trim(), 8);
+                        resultColor.A = channelbytes[0];
+                        resultColor.R = channelbytes[1];
+                        resultColor.G = channelbytes[2];
+                        resultColor.B = channelbytes[3];
+                        break;
+                    }
+                case ColorStringTypes.Hex_shortString:
+                    {
+                        byte[] channelbytes = ByteHexStringValues(input.Trim(), 6);
+                        resultColor.A = 255;
+                        resultColor.R = channelbytes[0];
+                        resultColor.G = channelbytes[1];
+                        resultColor.B = channelbytes[2];
+                        break;
+                    }
+                case ColorStringTypes.None:
+                    throw new TexParseException($"The text: {input} could not be parsed.");
+                default:
+                    break;
+            }
+            return resultColor;
+        }
+
+        #region Color Parsing Helpers
+        /// <summary>
+        /// Specifies the type of <see cref="Color"/> the string might represent.
+        /// </summary>
+        private enum ColorStringTypes
+        {
+            /// <summary>
+            /// Type is a <see cref="string"/> consisting of byte values of the color.
+            /// </summary>
+            Byte_longString,
+            /// <summary>
+            /// Type is a <see cref="string"/> consisting of byte values of the color.<para/>
+            /// Alpha channel is strictly = 255.
+            /// </summary>
+            Byte_shortString,
+            //FloatString,----->Will include this later.
+            /// <summary>
+            /// Type is a Hexadecimal <see cref="string"/> consisting of Hexadecimal values of the color.
+            /// </summary>
+            Hex_longString,
+            /// <summary>
+            /// Type is a Hexadecimal <see cref="string"/> consisting of Hexadecimal values of the color.<para/>
+            /// Alpha channel is strictly = FF.
+            /// </summary>
+            Hex_shortString,
+
+
+            /// <summary>
+            /// An unrecognized ColorStringType.
+            /// </summary>
+            None
+        }
+
+        /// <summary>
+        /// Gets the type of <see cref="ColorStringTypes"/> from the <paramref name="input"/>.
+        /// </summary>
+        /// <param name="input"></param>
+        /// <returns></returns>
+        private static ColorStringTypes GetColorString(string input)
+        {
+            if (IsByteHexTrain(input,6)==true)
```

Comment:
These numbers, 6, 8, 3 and 4, seems a little bit magical here. Could we have them in named constants?